src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -1290,6 +1294,57 @@ def __print_kvset_json_diff(old_obj=None, new_obj=None, yes=False):
     return True
 
 
+def __tags_equals(tag1, tag2):
+
+    if not tag1 and not tag2:
+        return True
+
+    if not (isinstance(tag1, dict) and isinstance(tag2, dict)):
+        return False
+
+    if len(tag1) != len(tag2):
+        return False
+
+    try:
+        for tag_key in tag1.keys():
+            if tag1[tag_key] != tag2[tag_key]:
+                return False
+
+        return True
+
+    except KeyError:
+        return False
+
+
+# Compare the key-value fields to determine whether two key-values are the same.
+# We use the specified label and content_type values for key-values being imported.
+# Feature flag content-type is constant so we use this for feature flag comparison instead of the input content_type argument.
+def __kv_equals(source_kv, dest_kv, input_label, input_content_type):
+    if not (
+            source_kv.key == dest_kv.key and
+            input_label == dest_kv.label and
+            __tags_equals(source_kv.tags, dest_kv.tags)):
+        return False
+
+    if source_kv.content_type == FeatureFlagConstants.FEATURE_FLAG_CONTENT_TYPE:
+        if source_kv.content_type != dest_kv.content_type:
+            return False
+
+        differ = JsonDiffer()
+        return differ.diff(json.loads(source_kv.value), json.loads(dest_kv.value)) == {}
```

Comment:
>source_kv.value), json.loads(dest_kv.value [](http://example.com/codeflow?start=38&length=42)

Why do we need to json diff the value? Can we directly compare both string values?

Reply:
There were situations where directly comparing the strings resulted in differences in the event where something there was, say, an extra space after a comma between two filters. Technically, such a feature flag is still the same but will be treated the same but will be detected to have changed. The reason for using a differ was to parse and find the difference between the actual json values.